packages/runtime/lifecycle/src/api.ts

```diff
@@ -32,33 +45,76 @@ export type ApplyInputsRequest = {
+   * @param request Request parameters for creating a Marlowe Contract on Cardano
    * @throws DecodingError
    */
-  create(req: CreateRequest): Promise<ContractId>;
-  // [[apply-inputs-next-provider]]
-  // DISCUSSION: Instead of providing a function from "Next" => ApplyInputsRequest, I think
-  //             this as a low level API should just receive the ApplyInputsRequest, and if we want we can
-  //             add a "higher-level" helper
-  //             Search for [apply-inputs-next-provider] for usage
+  submitCreateTx(request: CreateRequest): Promise<[ContractId, TxId]>;
+
   /**
-   * TODO: comment
+   * @experimental
+   * Submit to the Cardano Ledger & Wait Confirmation, the Transaction(Tx) that will create the Marlowe Contract passed in the request.
+   * It's higher level capabilities compared to `submitCreateTx`.
+   * It combines `submitCreateTx` & waitConfirmation for you to abstract Cardano infrastructure concerns.
+   * @param request Request parameters for creating a Marlowe Contract on Cardano
+   * @throws DecodingError
+   */
+  create(request: CreateRequest): Promise<ContractId>;
+
+  /**
+   * Submit to the Cardano Ledger, the Transaction(Tx) that will apply inputs to a given created contract (see `submitCreateTx`).
+   * It's lower level version of `applyInputs`.
+   * @param contractId Contract Id where inputs will be applied
+   * @param inputsRequest inputs to apply
+   * @throws DecodingError
+   */
+  submitApplyInputsTx(
+    contractId: ContractId,
+    inputsRequest: ApplyInputsRequest
+  ): Promise<TxId>;
+
+  /**
+   * Provide Applicable Inputs and Reducibility Information for a given contract for the connected wallet.
+   * @param contractId Contract Id of a created contract
+   * @throws DecodingError
+   */
+  getNext(contractId: ContractId): Promise<Next>;
+
+  /**
+   * @experimental
+   * Submit to the Cardano Ledger & Wait Confirmation, the Transaction(Tx) that will apply inputs to a given created contract (see `create`).
+   * It's higher level version of `submitApplyInputsTx`.
+   * It combines `submitApplyInputsTx`, `getNext` & waitConfirmation for you to abstract Cardano infrastructure concerns
+   * and gives you the ability to describe the execution of a Marlowe Contract in a more declarative way.
+   * @param contractId Contract Id where inputs will be applied
+   * @param provideInput It's a function to provide where the current `Next` of the given `contractId` is feeded
+   *                     and some applicable inputs needs to be returned.
    * @throws DecodingError
    */
   applyInputs(
     contractId: ContractId,
     provideInput: ProvideInput
-  ): Promise<ContractId>;
+  ): Promise<TxId>;
+
+  /**
+   * Waits for a transaction to be confirmed.
+   * @experimental
+   * @param txId the transaction hash/id to wait for confirmation
+   * @returns true if the transaction is confirmed, false otherwise
+   */
+
+  waitConfirmation(txId: TxId): Promise<boolean>;
```

Comment:
Any particular reason to re-export this here? Does it aggregate something to the wallet API?

Reply:
yeah, I was thinking it may not be intuitive enough if you need to go on WalletAPI to do it... The API is not self sufficient somehow...